### PR TITLE
Add link to amphtml url

### LINKED ISSUE #1337: Ensure canonical articles specify amp articles in metadata
**Is your feature request related to a problem? Please describe.**
Canonical articles don't currently specify the location of the equivalent AMP article.

**Describe the solution you'd like**
```
<link rel="amphtml" href="https://www.bbc.com/news/articles/c0000000000o.amp">
```
The TLD should always respect the same TLD as the user is on.

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Will need unit tests. Cypress tests could be dificult due to restrictions on tld due to mozart redirect on test. Investigation needed.

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by sareh**
Resolves #1337 

**Overall change:** Adds amphtml link tag to head of document, for the canonical pages only.

**Code changes:**

- Adds amphtml link tag to component
- Canonical link & og:url meta content always of format `${bbcOrigin}/${service}/articles/${id}`
- AMP HTML link always of format `${bbcOrigin}/${service}/articles/${id}.amp`

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/Metadata/index.jsx [hunk 1]
```diff
@@ -45,6 +45,7 @@ const MetadataContainer = ({ metadata, promo }) => {
           }) => {
             /* Canonical link generated from servicename and id */
             const canonicalLink = `https://www.bbc.com/${service}/articles/${id}`;
+            const ampLink = `https://www.bbc.com/${service}/articles/${id}.amp`;
```
Comment (ChrisBAshton) [inline src/app/containers/Metadata/index.jsx]: Can we start using the TLD Context for this?

File: cypress/integration/ampSpec.js [hunk 2]
```diff
@@ -71,6 +75,18 @@ describe('AMP Tests on a .amp page', () => {
     });
   });
 
+  it('should include the canonical URL & ampHTML', () => {
+    const defaultOrigin = 'https://www.bbc.co.uk';
```
Comment (sareh) [inline cypress/integration/ampSpec.js:75]: The default origin value is used for AMP pages.

File: cypress/integration/metaNewsSpec.js [hunk 3]
```diff
@@ -98,14 +99,22 @@ describe('Article Meta Tests', () => {
     metadataAssertion();
   });
 
-  it('should include the canonical URL', () => {
+  it('should include the canonical URL & ampHTML', () => {
+    const currentOrigin = window.location.origin;
```
Comment (sareh) [inline cypress/integration/metaNewsSpec.js:103]: For canonical pages, the current origin on the client is used.

File: cypress/integration/metaNewsSpec.js [hunk 4]
```diff
@@ -98,14 +99,22 @@ describe('Article Meta Tests', () => {
...
+    checkAmpHTML(
+      `${currentOrigin}/news/articles/${
         config.assets.newsThreeSubheadlines
-      }`,
+      }.amp`,
     );
   });
 
+  it('should include metadata that matches the JSON data', () => {
+    metadataAssertion();
```
Comment (pjlee11) [inline cypress/integration/metaNewsSpec.js]: Is the metadata change meant to be part of this PR?

File: src/app/containers/Metadata/index.jsx [hunk 5]
```diff
@@ -43,26 +43,25 @@ const MetadataContainer = ({ metadata, promo }) => {
             publishingPrinciples,
             noBylinesPolicy,
           }) => {
-            /* Canonical link generated from servicename and id */
-            const canonicalLink = `https://www.bbc.com/${service}/articles/${id}`;
+            const canonicalLink = `${origin}/${service}/articles/${id}`;
```
Comment (pjlee11) [inline src/app/containers/Metadata/index.jsx:46]: If this is using `origin` and we expect amp to [use the `default`](https://github.com/bbc/simorgh/pull/1497/files#r270533183) does this not break the expectations?

File: cypress/integration/metaNewsSpec.js (hunk 4 above)
Comment (sareh) [reply cypress/integration/metaNewsSpec.js]: Good point - a duplicate test snuck in. Removed it now

File: src/app/containers/Metadata/index.jsx (hunk 5 above)
Comment (sareh) [reply src/app/containers/Metadata/index.jsx:46]: @pjlee11 By default, I mean that `origin` has a default value of `https://www.bbc.co.uk` defined here: https://github.com/bbc/simorgh/blob/af741b24b53c3a9e40385bc50509424437fd3ee0/src/app/contexts/RequestContext/getOriginContext.js#L2

Comment (pjlee11) [reply src/app/containers/Metadata/index.jsx:46]: Ahhh okay that makes sense, thanks 👍

**Review by pjlee11 — DISMISSED** [review]
LGTM 👍

**Review by dr3 — DISMISSED** [review]

**Review by dr3 — DISMISSED** [review]

**Review by pjlee11 — APPROVED** [review]

**Review by dr3 — APPROVED** [review]
😱 force push

**Pull request merged**